Volume_08/Number_1/Guigue2003/tri_tri_intersect.c

```diff
@@ -65,6 +65,18 @@ int tri_tri_overlap_test_2d(double p1[2], double q1[2], double r1[2],
 
 
 
+/* Epsilon coplanarity checks */
+
+#define USE_EPSILON_TEST false /* set to true to use coplanarity robustness checks */
+#define EPSILON 1e-16          /* you may also need to adjust this value according to your data */
```

Comment:
Thanks for quickly adding a bit of guard code for this case. I see [Tomas's epsilon is larger](https://fileadmin.cs.lth.se/cs/Personal/Tomas_Akenine-Moller/code/opttritri.txt), I guess because he's using floats. Like you say, this might need to be adjusted anyway. I think it's fine as is as a way to let the user know that there's a value to be aware of.